concordium-consensus/src/Concordium/Skov/Update.hs

```diff
@@ -479,87 +480,116 @@ doStoreBlock pb@GB.PendingBlock{..} = unlessShutDown $ do
 -- the slot number of the block that the transaction was received with,
 -- and 0 if the transaction was received separately from a block.
 -- This returns
---
 --   * 'ResultSuccess' if the transaction is freshly added.
+--     The transaction is added to the transaction table.
 --   * 'ResultDuplicate', which indicates that either the transaction is a duplicate
+--     The transaction is not added to the transaction table.
 --   * 'ResultStale' which indicates that a transaction with the same sender
 --     and nonce has already been finalized, or the transaction has already expired. In this case the transaction is not added to the table.
 --   * 'ResultInvalid' which indicates that the transaction signature was invalid.
+--     The transaction is not added to the transaction table.
 --   * 'ResultShutDown' which indicates that consensus was shut down, and so the transaction was not added.
---   * 'ResultExpiryTooLate' which indicates that transaction's expiry was too far in the future and so the transaction
---     was not accepted.
 --   * 'ResultTooLowEnergy' which indicates that the transactions stated energy was below the minimum amount needed for the
 --     transaction to be included in a block. The transaction is not added to the transaction table
-doReceiveTransaction :: (TreeStateMonad pv m, TimeMonad m, SkovQueryMonad pv m) => BlockItem -> Slot -> m UpdateResult
+--   * 'ResultNonexistingSenderAccount' the transfer contained an invalid sender. The transaction is not added to the
+--     transaction table.
+--   * 'ResultDuplicateAccountRegistrationID' the 'CredentialDeployment' contained an already registered registration id.
+--     The transaction is not added to the transaction table.
+--   * 'ResultCredentialDeploymentInvalidSignatures' the 'CredentialDeployment' contained invalid signatures.
+--     The transaction is not added to the transaction table.
+--   * 'ResultCredentialDeploymentInvalidIP' the 'CredentialDeployment' contained an unreckognized identity provider.
+--     The transaction is not added to the transaction table.
+--   * 'ResultCredentialDeploymentInvalidAR' the 'CredentialDeployment' contained unreckognized anonymity revokers. 
+--     The transaction is not added to the transaction table.
+--   * 'ResultCredentialDeploymentExpired' the 'CredentialDeployment' was expired. The transaction is not added to
+--     the transaction table.
+--   * 'ResultChainUpdateInvalidSequenceNumber' the update contained an invalid 'UpdateSequenceNumber'.
+--   * 'ResultChainUpdateInvalidEffectiveTime' the update contained an invalid effective time. In particular the effective time
+--      was before the timeout of the update.
+--   * 'ResultChainUpdateInvalidSignatures' the update contained invalid signatures.
+--   * 'ResultEnergyExceeded' the stated energy of the transaction exceeds the maximum allowed for the block.
+doReceiveTransaction :: (TreeStateMonad pv m,
```

Comment:
I think the Slot argument to this function is never used with anything other than 0.

It is therefore confusing to have it for this function, so I'd recommend removing it.